src/RTCPeerConnection.ts

```diff
@@ -569,6 +615,42 @@ export default class RTCPeerConnection extends defineCustomEventTarget(...PEER_C
             }
 
             this.signalingState = ev.signalingState;
+            const { localSdp, remoteSdp } = ev;
+
+            if (localSdp.type && localSdp.sdp) {
+                switch (this.signalingState) {
+                    case 'stable':
+                        this._currentLocalDescription = new RTCSessionDescription(localSdp);
+                        this._pendingLocalDescription = null;
+                        break;
+                    case 'have-local-offer':
+                    case 'have-local-pranswer':
+                        this._currentLocalDescription = null;
```

Comment:
Isn't this block redundant with the handling in setLocalDescription?

Reply:
I’d rather say handling it in `setLocalDescription` is redundant, but I’ve left it there just in case SDP updated while 

```js
await WebRTCModule.peerConnectionSetLocalDescription(this._pcId, desc);
```

resolved.

Assume we call `setLocalDescription(offer)`. The fact is that we receive `onsignalingstatechange(have-local-offer)` event earlier than above promise resolved. So if we don't set current/pending SDP here, in client's JS code `peerConnection.pendingLocalDescription` would return `null`. 

Of course, we could handle all this stuff in `set(Local|Remote)Description`, but it would make us wrap 
```js
await WebRTCModule.peerConnectionSetLocalDescription(this._pcId, desc);
```
in try catch and restore SDPs in case of exception. I think in current way code looks more concise.

Reply:
Makes sense, thanks for th explanation!

In that case let's remove the handling here so it's centralized in the other place?

Reply:
Do you mean remove from `peerConnectionSignalingStateChanged` listener and move everything to the `set(Local|Remote)Description`?

Reply:
The opposite, as you suggested